### PR TITLE
Bump Gorm and related dependencies

### PR DESCRIPTION
**Pull request by alan-ghelardi**
Resolves https://github.com/tektoncd/results/issues/273.

##  What

This pull request is a cherry-pick of
https://github.com/tektoncd/results/pull/300, including the bump of the
following gorm dependencies to fix compilation errors:

-       gorm.io/driver/postgres to v1.4.6.
- gorm.io/driver/sqlite to v1.4.4.
- github.com/mattn/go-sqlite3 to v1.14.16.

See the comment added by dependabot on the gorm.io/gorm's bump:

Bumps [gorm.io/gorm](https://github.com/go-gorm/gorm) from 1.23.4 to 1.24.3.
- [Release notes](https://github.com/go-gorm/gorm/releases)
- [Commits](https://github.com/go-gorm/gorm/compare/v1.23.4...v1.24.3)

---
cccupdated-dependencies:
- dependency-name: gorm.io/gorm
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

## Issues

This bump reveals an issue with the database auto-migration performed on the API
server's startup, which is conflicting with the schema installed by the
results.sql file. We cannot merge this pull request until this issue isn't
resolved. I'm going to open a separate issue to track this problem.

### CODE REVIEW

**Comment by alan-ghelardi** [comment]
Running tests to illustrate the failures in the database auto-migration.
/test all

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
 	github.com/jackc/pgconn v1.13.0
 	github.com/jonboulle/clockwork v0.3.0
-	github.com/mattn/go-sqlite3 v2.0.3+incompatible
+	github.com/mattn/go-sqlite3 v1.14.16
```
Comment (adambkaplan) [inline go.mod:20]: Is this technically a downgrade?

**Review by adambkaplan — COMMENTED** [review]
/approve

One question about `go-sqlite3`, otherwise looks good.

File: go.mod (hunk 1 above)
Comment (alan-ghelardi) [reply go.mod:20]: Actually the latest version is 1.14.16 indeed. The v2 was released by mistaken. From their docs:

>Latest stable version is v1.14 or later, not v2. NOTE: The increase to v2 was an accident. There were no major changes or features.

This bump is necessary to fix compilation errors that came out from the sqlite bump.

**Comment by adambkaplan** [comment]
LGTM - pending rebase other folks can add the label and this can merge.

**Comment by alan-ghelardi** [comment]
@khrm  could you second this one please?

**Review by khrm — COMMENTED** [review]
/lgtm

**Review by khrm — COMMENTED** [review]
/kind cleanup
We need this label for releasing as discussed in the last WG call.

**Pull request merged**